app/scripts/views/sign_in_totp_code.js

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const AuthErrors = require('lib/auth-errors');
+  const Cocktail = require('cocktail');
+  const Constants = require('../lib/constants');
+  const FormView = require('./form');
+  const SignInMixin = require('./mixins/signin-mixin');
+  const ServiceMixin = require('./mixins/service-mixin');
+  const Template = require('templates/sign_in_totp_code.mustache');
+  const VerificationReasonMixin = require('./mixins/verification-reason-mixin');
+
+  const CODE_INPUT_SELECTOR = 'input.totp-code';
+
+  const View = FormView.extend({
+    className: 'sign-in-totp-code',
+    template: Template,
+
+    _santizeCode (code) {
+      // Remove spaces and `-`
+      return code.replace(/[- ]*/g, '');
+    },
+
+    getAccount () {
+      return this.getSignedInAccount();
+    },
+
+    beforeRender () {
+      // user cannot confirm if they have not initiated a sign in.
+      if (! this.getAccount()) {
+        this.navigate(this._getAuthPage());
```

Comment:
hm I can still navigate to the page and get /missing sessiontoken ( http://127.0.0.1:3030/signin_totp_code ) . Is this not working?

<img src=https://user-images.githubusercontent.com/128755/37054538-23fb2f5e-214d-11e8-8daa-1eb937ed947a.png width=300 />